src/video/emscripten/SDL_emscriptenevents.c

```diff
@@ -1005,31 +1005,17 @@ static void Emscripten_unset_drag_event_callbacks(SDL_WindowData *data)
     }, data->canvas_id);
 }
 
-static const char *Emscripten_GetKeyboardTargetElement()
+static bool Emscripten_ShouldSetKeyboardElement(SDL_WindowData* data)
 {
-    const char *target = SDL_GetHint(SDL_HINT_EMSCRIPTEN_KEYBOARD_ELEMENT);
-
-    if (!target || !*target) {
-        return EMSCRIPTEN_EVENT_TARGET_WINDOW;
-    }
-
-    if (SDL_strcmp(target, "#none") == 0) {
-        return NULL;
-    } else if (SDL_strcmp(target, "#window") == 0) {
-        return EMSCRIPTEN_EVENT_TARGET_WINDOW;
-    } else if (SDL_strcmp(target, "#document") == 0) {
-        return EMSCRIPTEN_EVENT_TARGET_DOCUMENT;
-    } else if (SDL_strcmp(target, "#screen") == 0) {
-        return EMSCRIPTEN_EVENT_TARGET_SCREEN;
+    if (data->keyboard_element_is_string) {
+        return SDL_strcmp(data->keyboard_element, "#none") != 0;
+    } else {
+        return data->keyboard_target != EMSCRIPTEN_EVENT_TARGET_INVALID;
     }
```

Comment:
This undoes #12493, which is required since SDL hints/properties can't be set to any of the `EMSCRIPTEN_EVENT_TARGET_*` values: `SDL_SetHint()`/`SDL_SetStringProperty()` will try to `strdup()` the string which 1) is pointing to garbage so who knows what it'll contain, and 2) won't have the magic address matching the `EMSCRIPTEN_EVENT_TARGET_*` values.

Without those values being possible, `keyboard_element_is_string` doesn't need to exist either since it'll always be a valid string or `NULL`.

Reply:
My thought process was to allow users to use `EMSCRIPTEN_EVENT_TARGET_*` as the window property. So, using it as the default would ensure that its possible.

Reply:
But users can't set any hints/properties to `EMSCRIPTEN_EVENT_TARGET_*` since they get treated as real strings like I mentioned in my first comment.